core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -277,8 +277,7 @@ object SparkEnv extends Logging {
       "spark.serializer", "org.apache.spark.serializer.JavaSerializer")
     logDebug(s"Using serializer: ${serializer.getClass}")
 
-    val closureSerializer = instantiateClassFromConf[Serializer](
-      "spark.closure.serializer", "org.apache.spark.serializer.JavaSerializer")
+    val closureSerializer = new JavaSerializer(conf)
```

Comment:
I kept it just because I assume the usages of the closure serializer must use the Java serializer, and can't use the generic `spark.serializer` since that may be set to Kryo. Sure, there's probably not much point in passing it in as an arg, I can change that.